File: FormalBench-data/diverse/rule_14/FindPeak.java

import java.io.*;
import java.lang.*;
import java.math.*;

class FindPeak {

	public static int findPeak(int[] arr, int n) {
		int left = 0, right = n - 1;
		while (left < right) {
			int mid = left + (right - left) / 2;
			if (!(arr[mid] < arr[mid + 1])) {
				right = mid;
			} else {
				left = mid + 1;
			}
		}
		return left;
	}
}

File: FormalBench-data/FilteredData/translated_c/seed_726_per_category_10_653ade686f/FindPeak.c
#include "java_arrays.h"
#include <stdint.h>

static int32_t java_add(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left + (uint32_t)right);
}

static int32_t java_sub(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left - (uint32_t)right);
}

static int32_t java_div(int32_t dividend, int32_t divisor) {
    if (dividend == INT32_MIN && divisor == -1) {
        return INT32_MIN;
    }
    return dividend / divisor;
}

int32_t findPeak(JIntArray arr, int32_t n) {
    int32_t left = 0;
    int32_t right = java_sub(n, 1);
    while (left < right) {
        int32_t mid = java_add(left, java_div(java_sub(right, left), 2));
        int32_t next = java_add(mid, 1);
        if (jarray_get(arr, mid) < jarray_get(arr, next)) {
            left = next;
        } else {
            right = mid;
        }
    }
    return left;
}
